src/block/precomputed.rs

```diff
@@ -33,12 +34,14 @@ pub struct PrecomputedBlock {
     pub protocol_state_proof: ProtocolStateProofV1,
     pub staged_ledger_diff: StagedLedgerDiff,
     pub delta_transition_chain_proof: DeltaTransitionChainProof,
+    pub global_slot_number: Option<u32>,
 }
 
 impl PrecomputedBlock {
     pub fn from_log_contents(log_contents: BlockLogContents) -> serde_json::Result<Self> {
         let state_hash = log_contents.state_hash;
         let blockchain_length = log_contents.blockchain_length;
+        let global_slot_number = log_contents.global_slot_number;
```

Comment:
Or you maybe you can grab it from `protocol_state`?

Reply:
@Isaac-DeFrain We could write `        let global_slot_since_genesis = log_contents.global_slot_since_genesis;
` after the `serde_json::from_str::<BlockLog>` but then we will get the error message `use of partially moved value: `consensus_state` partial move occurs because `consensus_state.global_slot_since_genesis` has type `U32Json`, which does not implement the `Copy` trait` and idk if we want to implement the Copy trait U32Json, which originates from the mina-rs dependecy / submodule.

Reply:
You can probably just `clone()` it